core/federated/federate.c

```diff
@@ -1932,8 +1933,26 @@ bool update_max_level(tag_t tag, bool is_provisional) {
         // Safe to complete the current tag
         return (prev_max_level_allowed_to_advance != max_level_allowed_to_advance);
     }
-    for (int i = 0; i < _lf_zero_delay_action_table_size; i++) {
-        lf_action_base_t* input_port_action = _lf_zero_delay_action_table[i];
+#ifdef FEDERATED_DECENTRALIZED
+    size_t action_table_size = _lf_action_table_size;
+    lf_action_base_t** action_table = _lf_action_table;
+#else
+    size_t action_table_size = _lf_zero_delay_action_table_size;
+    lf_action_base_t** action_table = _lf_zero_delay_action_table;
```

Comment:
Yes, there seems to be a subtle mismatch between terminologies. I have been using the term "zero" to mean "(0, 0)" because (0, 0) is an additive identity and have been considering "(0, 1)" to be nonzero because if you add it to something you get a different logical time (that is one microstep later).